dimod/binary_quadratic_model/model.py

```diff
@@ -0,0 +1,1099 @@
+
+        Args:
+            v (variable):
+                A variable in the binary quadratic model.
+
+        Notes:
+            If the given variable is not in the binary quadratic model, this function does nothing.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel({'a': 1., 'b': 2.}, {}, 0.0, dimod.SPIN)
+            >>> bqm.remove_variable('a')
+            >>> bqm.linear
+            {'b': 2.}
+
+        """
+        linear = self.linear
+        if v in linear:
+            del linear[v]
+        else:
+            # nothing to remove
+            return
+
+        quadratic = self.quadratic
+        adj = self.adj
+
+        for u in adj[v]:
+            if (u, v) in quadratic:
+                del quadratic[(u, v)]
+            else:
+                del quadratic[(v, u)]
+
+            del adj[u][v]
+
+        del adj[v]
+
+        try:
+            # invalidates counterpart
+            del self._counterpart
+            if hasattr(self, '_binary'):
+                del self._binary
+            if hasattr(self, '_spin'):
+                del self._spin
+        except AttributeError:
+            pass
+
+    def remove_variables_from(self, variables):
+        """Remove the given variables and all of their interactions.
+
+        Args:
+            variables(iterable):
+                A collection of variables to be removed from the binary quadratic model.
+
+        Notes:
+            If any variable is not in the binary quadratic model, it is ignored.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel({'a': 1., 'b': 2., 'c': 3.}, {}, 0.0, dimod.SPIN)
+            >>> bqm.remove_variables_from(['a', 'c'])
+            >>> bqm.linear
+            {'b': 2.}
+
+        """
+        for v in variables:
+            self.remove_variable(v)
+
+    def remove_interaction(self, u, v):
+        """Remove the interaction between u, v.
+
+        Args:
+            u (variable):
+                A variable in the binary quadratic model that has an interaction with v.
+
+            v (variable):
+                A variable in the binary quadratic model that has an interaction with u.
+
+        Notes:
+            Any interaction not in the binary quadratic model is ignored.
+
+        Examples:
+            >>> bqm = dimod.BinaryQuadraticModel({}, {('a', 'b'): -1, ('b', 'c'): 1}, 0.0, dimod.SPIN)
+            >>> bqm.remove_interaction('b', 'c')
+            >>> bqm.quadratic
+            {('a', 'b'): -1}
+            >>> bqm.remove_interaction('a', 'c')  # not an interaction, so ignored
+            >>> bqm.quadratic
+            {('a', 'b'): -1}
+
+        """
+        quadratic = self.quadratic
+        adj = self.adj
+
+        try:
+            del adj[v][u]
+        except KeyError:
+            return  # no interaction with that name
+        del adj[u][v]
+
+        if (u, v) in quadratic:
+            del quadratic[(u, v)]
+        if (v, u) in quadratic:
```

Comment:
Maybe make this an else so that an error is raised if there somehow was an inconsistency (like line 510).

Reply:
Good idea, fixed